.github/workflows/tests.yml

```diff
@@ -16,7 +16,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```

Comment:
I would only add 3.11 and 3.12 and not remove 3.7 (maybe because EOL) and 3.8.

Reply:
The `leading_edge` job either way only runs on py 3.12 and numpy 2 wont be available for py 3.8 and lower.

Reply:
I'm happy to test 3.9-3.12 as implemented in this PR. Thanks!!